src/view/placeholder.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * @license Copyright (c) 2003-2017, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/**
+ * @module engine/view/placeholder
+ */
+
+import extend from '@ckeditor/ckeditor5-utils/src/lib/lodash/extend';
+import EmitterMixin from '@ckeditor/ckeditor5-utils/src/emittermixin';
+import CKEditorError from '@ckeditor/ckeditor5-utils/src/ckeditorerror';
+
+const CSS_CLASS = 'ck-placeholder';
+const DATA_ATTRIBUTE = 'data-placeholder';
```

Comment:
I remember I did something like this in my first PR to the CKEditor 4. And I remember that Fred gave me R- then telling it an unnecessary complication. Now, I see that we use it more and more. And to some point I like it.

But on the other hand, it's 2017 and all of us have IDEs which will easly find for us all `data-placeholder` and replace with anything we want. Also, I think that if one day we will want to replace it with something else, it probably will be a configurable variable so it will have to be changed anyway. At the same time, the fact that I have `data-placeholder` in the Chrome dev tools, and `DATA_ATTRIBUTE` in the code, and need to check and remember that it equals does not help to debug. It might not be hard with one variable, but we get more and more of these. I know, that it's a popular good practice, but, my question is: what do we get?